### PR TITLE
SimulationConfig parse target_simulator, node_sets_file and node_set

### PR DESCRIPTION
**Pull request by WeinaJi**

### CODE REVIEW

File: src/config.cpp [hunk 1]
```diff
@@ -827,10 +832,41 @@ class SimulationConfig::Parser
...
+            return toAbsolute(_basePath, val);
+        } else {
+            try {
+                const auto circuitFile = parseNetwork();
+                const auto conf = CircuitConfig::fromFile(circuitFile);
+                return conf.getNodeSetsPath();
+            } catch (...) {
+                // Don't throw CircuitConfig exceptions in SimulationConfig and return empty string
+                return val;
+            }
```
Comment (WeinaJi) [inline src/config.cpp:858]: To be discussed with @mgeplf @ferdonline @jorblancoa 
1) The param `node_sets_file` is to override the one from circuit_config file, but it is optional in circuit_config with the default of empty string. What if no `node_sets_file` in circuit_config nor in simulation_config? Does SimulationConfig return empty string or `"node_sets.json"`? 
2) Here I set no throw, that SimulationConfig parser does not check the validity of the circuit config file.

Comment (mgeplf) [reply src/config.cpp:906]: Since we're using `optional` in several places, perhaps that's the way to go?

Comment (WeinaJi) [reply src/config.cpp:906]: Yes, I agree. Then the default value is also an empty string in SimulationConfig. Also in the case that `node_set` is not provided (meaning all non-virtual populations),  `node_sets_file` is not needed.

**Comment by WeinaJi** [comment]
Hello, any more comment for this PR? Can we merge it ?

File: src/config.cpp (hunk 1 above)
Comment (mgeplf) [reply src/config.cpp:906]: Probably worth getting other people's opinions, I think.

Comment (WeinaJi) [reply src/config.cpp:906]: @jorblancoa @ferdonline @NadirRoGue , what do you think about the two items above?

Comment (NadirRoGue) [reply src/config.cpp:906]: I prefer an empty string (or nullopt) to signal there is no file and to be consistent with CircuitConfig

Comment (WeinaJi) [reply src/config.cpp:906]: I have discussed with @ferdonline @jorblancoa offline, and they are both fine with the empty string as the default `node_sets_file`. Jorge raised the point that `reports:cells` should be optional as well, for which I opened a MR for discussion
https://bbpgitlab.epfl.ch/common/doc/circuit-documentation/-/merge_requests/24

For this PR, I think we can go ahead to merge.

Comment (mgeplf) [reply src/config.cpp:906]: > For this PR, I think we can go ahead to merge.

Ok, will do.

**Review by mgeplf — APPROVED** [review]

**Pull request merged**